classes/publication/maps/Schema.inc.php

```diff
@@ -35,7 +36,12 @@ protected function mapByProperties(array $props, Publication $publication, bool
                         function ($chapterAuthor) use ($publication) {
                             return $chapterAuthor->_data;
                         },
-                        DAORegistry::getDAO('ChapterAuthorDAO')->getAuthors($publication->getId(), $chapter->getId())->toArray()
+                        Repo::author()->getMany(
+                            Repo::author()
+                                ->getCollector()
+                                ->filterByChapter($chapter->getId())
+                                ->filterByPublicationIds([$publication->getId()])
+                        )->toArray()
```

Comment:
Now that it's returning a `LazyCollection`, you can't run `array_map` on it directly. Instead you should use the collection's `map` method: https://laravel.com/docs/8.x/collections#method-map

That would look like this:

```php
$data['authors'] = Repo::author()
    ->getMany(...)
    ->map(function($chapterAuthor) {
        return $chapterAuthor->_data;
    });
```

Reply:
Fixed I think